mep/accounts/management/commands/twitterbot_100years.py

```diff
@@ -0,0 +1,264 @@
+        else:
+            # by default, report relative to today
+            # determine date 100 years earlier
+            relative_date = datetime.date.today() - relativedelta(years=100)
+
+        return relative_date
+
+    def find_events(self, date):
+        '''Find events 100 years before the current day or
+        a specified day in YYYY-MM-DD format.'''
+
+        # find all events for this date
+        # exclude partially known dates
+        # - purchase date precision == start date precision
+        # (borrow end *could* have different precision than start date)
+        events = Event.objects \
+            .filter(Q(start_date=date) |
+                    Q(subscription__purchase_date=date) |
+                    Q(borrow__isnull=False, end_date=date)) \
+            .filter(Q(start_date_precision__isnull=True) |
+                    Q(start_date_precision=int(self.full_precision)))
+        return events
+
+    def report(self, date):
+        # needs two modes:
+        # - schedule
+        # - tweet (takes event id)
+        for ev in self.find_events(date):
+            print('event type %s' % ev.event_type)
+            print('account %s' % ev.account)
+            print(ev)
+            content = tweet_content(ev, date)
+            print(content)
+
+    # times:  9 AM, 12 PM, 1:30 PM, 3 PM, 4:30 PM, 6 PM, 8 PM
+    tweet_times = ['9:00', '12:00', '13:30', '15:00', '16:30', '18:00',
+                   '20:00']
+
+    def schedule(self, date):
+        # find all events for today
+        self.find_events(date)
+        # filter out any that can't be tweeted
+        events = [ev for ev in self.find_events(date) if can_tweet(ev, date)]
+
+        # schedule the ones that can be tweeted
+        for i, ev in enumerate(events):
+            self.tweet_at(ev, self.tweet_times[i])
+
+    def tweet_at(self, event, time):
+        '''schedule a tweet for later today'''
+        cmd = '%s %s/manage.py twitterbot_100years tweet --event %s' % \
+            (sys.executable, settings.PROJECT_ROOT, event.id)
+        print('scheduling %s at %s' % (event, time))
+        print(cmd)
+        result = subprocess.run(['/usr/bin/at', time], input=cmd.encode(),
+                                stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        print(result.stderr.decode())
+
+    def tweet(self, event, date):
+        # make sure the event is tweetable
+        if not can_tweet(event, date):
+            return
+
+        content = tweet_content(event, date)
+        if not content:
+            return
+        print(content)
+        # TODO: set credentials in local settings
+        # auth = tweepy.OAuthHandler('', '')
+        # auth.set_access_token('', '')
+        # api = tweepy.API(auth)
+        # api.update_status(content)
+
+
+tweetable_event_types = ['Subscription', 'Renewal', 'Reimbursement',
+                         'Borrow', 'Purchase', 'Request']
```

Comment:
whoa, what's a Request?

Reply:
It's one of the new custom types that can be set based on notation in the private notes!